src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -104,16 +70,25 @@ func (a *App) onBootstrapWorkerNode(ctx context.Context, s *state.State, encoded
 	}
 	// TODO(neoaggelos): figure out how to use the microcluster client instead
 
-	// Get remote certificate from the cluster member
-	cert, err := utils.GetRemoteCertificate(token.JoinAddresses[0])
+	// Get remote certificate from the cluster member. We only need one node to be reachable for this.
+	// One might fail because the node is not part of the cluster anymore but was at the time the token was created.
+	var cert *x509.Certificate
+	var address string
+	var err error
+	for _, address = range token.JoinAddresses {
+		cert, err = utils.GetRemoteCertificate(address)
+		if err == nil {
+			break
+		}
+	}
```

Comment:
In general, try to avoid mixing dependency upgrades along with changes in the code logic, as it would be harder to backport and/or revert if required.